### ISSUE TITLE
Use findAndModify to avoid race conditions

### ISSUE THREAD

**Issue by lautarodragan** [poetapp/node#328]
Running multiple instances of the node behind a load balancer could lead to race conditions.

https://github.com/poetapp/node/blob/b1ffa5e6ec7a0de6951c13794f5421dcd68920c1/src/Storage/ClaimController.ts#L111-L117

To ensure there are no race conditions we need to use [findAndModify](https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify) and ensure the [proper indexes are created](https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify/#upsert-and-unique-index).

**Comment by geoffturk** [comment]
Assigning to @warrenv.

**Comment by WesleyCharlesBlake** [comment]
@lautarodragan  and @wzalazar : would we need to replicate this across to frost-api as well?

**Comment by lautarodragan** [comment]
@WesleyCharlesBlake we already have four instances of Frost API behind an LB in production and they have been working fine :sunglasses: 

Frost API has much less state in general (since it delegates most operations to the node), but it'd be nice to do a thorough check just to be on the safe side.

**Comment by WesleyCharlesBlake** [comment]
Awesome! Was just making sure. But yes will need a thorough audit in frost-api . So for in k8s it's been working well.

**Comment by geoffturk** [comment]
Reassigning to @kennylavender. Please sync with Warren if you need any input.

**Comment by WesleyCharlesBlake** [comment]
yeah simple enough to test out, just need to increase the replica count for each node configuration from 1 to n (where n is the ideal amount needed to run optimally)
